### PR TITLE
added VSAC api support to fetch the latest profile for a program

### PR DESCRIPTION
**Pull request by hossenlopp**
Adds support to fetch the latest profile for a given program.

Pull requests into Bonnie Bundler require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1502 https://jira.mitre.org/browse/BONNIE-1507
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here: There a single line of code for catching a 404 response from the server for invalid profile. This doesn't happen right now and I couldn't easily make a test for that situation, but it is included incase VSAC fixes that in the future.
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:

| branch | coverage |
| - | - |
| master | 804 / 1261 LOC (63.76%) |
| vsac_latest_profile_bonnie-1502 | 813 / 1271 LOC (63.97%) |

- [x] Automated regression test(s) pass N/A no calculation code changed

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint N/A

**Reviewer 1:**

Name: @mayerm94 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree N/A

**Reviewer 2:**

Name: @c-monkey
- [X] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [X] The tests appropriately test the new code, including edge cases
- [X] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree N/A

### CODE REVIEW

**Review by c-monkey — DISMISSED** [review]
looks good; tests pass

File: lib/util/vsac_api.rb [hunk 1]
```diff
@@ -171,6 +171,38 @@ def get_program_details(program = nil)
...
+      #
+      # Returns the name of the latest profile for the given program.
+      def get_latest_profile_for_program(program = nil)
+        # if no program was provided use the one in the config or default in constant
+        if program.nil?
+          program = @config.fetch(:program, DEFAULT_PROGRAM)
+        end
+
+        begin
+          # parse json response and return it
```
Comment (mayerm94) [inline lib/util/vsac_api.rb:194]: Suggest copying call description from VSAC API documentation "Retrieve the program information of a given program, including the list of releases." Noting that we are only interested in the name field returned

Comment (hossenlopp) [reply lib/util/vsac_api.rb:194]: There is no documentation for this call. It only returns the latest profile and a timestamp of the request.

`https://vsac.nlm.nih.gov/vsac/program/CMS eCQM/latest profile`
returns
```
{
    "name": "eCQM Update 2018-05-04",
    "requestTime": "2018-05-21 03:39:04 PM"
}
```

Comment (mayerm94) [reply lib/util/vsac_api.rb:194]: Wow, was only looking at the program call in the API docs, oops. I still think the comment should be updated to describe what we are expecting the be returned by the API here

Comment (hossenlopp) [reply lib/util/vsac_api.rb:194]: Added to comment.

**Review by mayerm94 — APPROVED** [review]

**Review by c-monkey — APPROVED** [review]
Tests pass. New comment is good.
